libhb/work.c

```diff
@@ -1415,6 +1442,50 @@ static void sanitize_filter_list(hb_list_t *list)
             }
         }
     }
+
+    int is_detel = 0;
+    hb_filter_object_t * filter = hb_filter_find(list, HB_FILTER_DETELECINE);
+    if (filter != NULL)
+    {
+        is_detel = 1;
+    }
+
+    filter = hb_filter_find(list, HB_FILTER_VFR);
+    if (filter != NULL)
+    {
+        int mode = hb_dict_get_int(filter->settings, "mode");
+        // "Same as source" FPS and no HB_FILTER_DETELECINE
+        if ( (mode == 0) || (is_detel = 0) )
+        {
+            hb_list_rem(list, filter);
+            hb_filter_close(&filter);
+            hb_log("Skipping VRF filter");
+        }
+    }
+    
+    filter = hb_filter_find(list, HB_FILTER_CROP_SCALE);
+    if (filter != NULL)
+    {
+        hb_dict_t* settings = filter->settings;
+        if (settings != NULL)
+        {
+            int width, height, top, bottom, left, right;
+            hb_dict_extract_int(&width, settings, "width");
+            hb_dict_extract_int(&height, settings, "height");
+            hb_dict_extract_int(&top, settings, "crop-top");
+            hb_dict_extract_int(&bottom, settings, "crop-bottom");
+            hb_dict_extract_int(&left, settings, "crop-left");
+            hb_dict_extract_int(&right, settings, "crop-right");
```

Comment:
hb_dict_extract_\* is meant to be used where you need to know if the key exists in the dictionary.  It returns 1 if the key was found and 0 if not.  It leaves the destination value unchanged if the key did not exist.  I would use hb_dict_get_int here (as you did above) because it will return a zero value if the key does not exist.

Reply:
Thanks for your feedback! I will split this pull request into two separate pull requests: general default filter behavior and QSV filters behavior!